extension/src/Xliff/XliffCache.ts

```diff
@@ -0,0 +1,30 @@
+import { Xliff } from "./XLIFFDocument";
+import * as path from "path";
+
+const cachedXliffDocuments: Map<string, Xliff> = new Map();
+
+export function getXliffDocumentFromCache(filePath: string): Xliff {
+  const fileName = path.basename(filePath);
+  if (!xliffDocumentInCache(fileName)) {
+    const newXliffDocument = Xliff.fromFileSync(filePath);
+    cachedXliffDocuments.set(fileName, newXliffDocument);
+    return newXliffDocument;
+  }
+  const xliffDocument = cachedXliffDocuments.get(fileName);
+  if (xliffDocument) {
+    return xliffDocument;
+  }
+  throw new Error(`${fileName} not found.`);
+}
+export function updateXliffDocumentInCache(
+  filePath: string,
+  content: string
+): void {
+  const fileName = path.basename(filePath);
+  const newXliffDocument = Xliff.fromString(content);
+  cachedXliffDocuments.set(fileName, newXliffDocument);
+}
+
+function xliffDocumentInCache(fileName: string): boolean {
+  return cachedXliffDocuments.has(fileName);
+}
```

Comment:
Just a thought; Would this work as a class?

Kind of like I did with BaseAppTranslationFiles. To tired to think of any true benefits in this implementation.

Reply:
Except maybe some state keeping down the line.

Reply:
Don't know really, then we would need to export the static instance of that class instead, I guess?

Maybe we should wait with this refactoring until we see the real benefits? It should be quite easy to refactor later, since it is already isolated.